### ISSUE TITLE
Async RwLock and Mutex.

### ISSUE THREAD

**Issue by weisrc** [monoio-rs/monoio#298]
The local sync library doesn't have an async RwLock and Mutex.

**Comment by ihciah** [comment]
Async RwLock and Mutex do not depend on async runtime. You can use any implementation. I think it's better to provide them in third-party libraries.
Note: if you want these mutex work under monoio, please enable `sync` feature, since these mutex require futures be able to woken from another thread.

**Comment by weisrc** [comment]
Ok, thanks. So I can use Tokio's Async RwLock and Mutex safely with Monoio?

**Comment by ihciah** [comment]
> So I can use Tokio's Async RwLock and Mutex safely with Monoio?

Yes. You can also use its channel implementation, or third-party crates like async-channel.
